### ISSUE TITLE
Improve mobile browsing with "calm" template.

### ISSUE THREAD

**Issue by Badatos** [websvnphp/websvn#91]
Hello !
Thank you for this useful app ;)
I have just a small remark very easy to correct : there is no "viewport" tag defined, so the mobile navigation is not optimal.
Maybe you can just add this line in the "head" section of your templates : 

`<meta name="viewport" content="width=device-width, initial-scale=1">`

thanks ;)

**Comment by ams-tschoening** [comment]
Could you please provide screenshots of what "not optimal" exactly means?

**Comment by Badatos** [comment]
Here they are : 

Without viewport:
![IMG_1825](https://user-images.githubusercontent.com/12731381/64530031-7a833c00-d30c-11e9-858a-b5b344a87061.png)
==> we have to scroll to the right, but we don't really know there is something right if we don't know websvn

With Viewport:
![IMG_1824](https://user-images.githubusercontent.com/12731381/64530123-adc5cb00-d30c-11e9-9de4-c7e9675e7ee9.png)
==> Almost all informations are here. It's not perfect and there are other things to do for responsivity, but it's a first step

**Comment by michael-o** [comment]
Did you manually change the viewport and see a difference?

**Comment by Badatos** [comment]
Yes, to see the difference I manually added the line in the `calm/header.tmpl` file.

**Comment by Badatos** [comment]
If you want something  really better on mobile, I just made some improvements on the calm/styles.css (in combination with the viewport tag) :
[styles.css](https://wimsedu.info/websvn/templates/calm/styles.css)
 You can test it directly on my server : https://wimsedu.info/websvn/?template=calm
(just reduce the window to a small size to see the difference)

**Comment by Badatos** [comment]
Hello !
Is still somebody there ? ;)

**Comment by ams-tschoening** [comment]
I'm here, but don't find this too important for me and simply didn't come to a conclusion yet if adding `viewport` is worth it. The layout obviously breaks after adding it without any additional media queries, while in the other case things seem to look fine, but one needs to scroll. If I test your website and my own in my current Opera using the device toolbar, I get the following results:

![viewport](https://user-images.githubusercontent.com/6223655/65156720-db5af480-da2f-11e9-80fc-c446d21bc53e.png)

![no viewport](https://user-images.githubusercontent.com/6223655/65156721-db5af480-da2f-11e9-8a7d-509a45b880e5.png)

While your text is clearly better readable, the layout is still broken and one can't even see file names or other aspects of the table properly. While the layout not using `viewport` looks a lot better, everything can be read, but without zooming it's pretty unusable and unreadable.

Adding `viewport` means our site is responsively designed and that it looks better and/or is better usable afterwards. But is that really the case? No doubt that the skins need to be enhanced for mobile devices, but maybe this should be done first and only afterwards `viewport` added?

**Comment by Badatos** [comment]
You're right. `viewport` is only a first step in mobile design, and it has to be combined with proper media queries.
I just made some adjustments to correct the issues in your screenshot, and here is what it's like now : 
<img width="439" alt="Capture d’écran 2019-09-18 à 16 50 11" src="https://user-images.githubusercontent.com/12731381/65159515-7ce44500-da34-11e9-9b8f-97e5415c6eff.png">

revjump is on its own line, and the table has it's own scroll bar, as recommended here : 
https://www.w3schools.com/howto/howto_css_table_responsive.asp

(i added a <div class="table-responsive"> in calm/directory.tmpl, line 73)

**Comment by ams-tschoening** [comment]
Changing here a bit, changing there a bit, for all current skins... I don't have the time to do so currently, so leave it to someone else. My current opinion is that `viewport` should not be added for itself only, which might make this issue be closed in favor of some general mobile skin-enhancement one or this one renamed or ...

@Badatos I suggest you simply fork the project, create a branch specially for supporting mobile devices, apply changes as you see fit and create a PR using that branch and linking to this discussion. This way you can easily maintain your setup and we don't need to reread the whole discussion for each and every single change mentioned as text only. At some point you might already have done all the work which can simply be merged. :-)

**Comment by Badatos** [comment]
The suggestion was not inevitably to be done on all skins yet, let's just start with calm, as the job is already done ^^.

**Comment by ams-tschoening** [comment]
I see this issue to be done, we focused on `calm` by purpose: Improving other skins not only means adding `viewport`, but also to improve broken rendering using media queries or stuff like that, somewhat like has been done in the PR.

In case of further improvements to `calm` or other skins, in my opinion it might be enough already to simply fork, change and create PRs describing what has actually been changed why. Screenshots might be a good idea, like has been done here. My point is that those specific things should be discussed at the specific PRs implementing the changes instead of some long, generic and verbose "improve"-issue. If things need discussion before actually changing things, feel free to create issues of course.
